database/models.go

```diff
@@ -0,0 +1,44 @@
+package database
+
+import (
+	"database/sql"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+type Election struct {
+	ID         uuid.UUID    `gorm:"primaryKey"`
+	Name       string       `gorm:"not null"`
+	Published  bool         `gorm:"not null"`
+	Finalized  bool         `gorm:"not null"`
+	OpenTime   sql.NullTime ``
+	CloseTime  sql.NullTime ``
+	Candidates []Candidate  `gorm:"many2many:candidate_in_election"`
+	Votes      []Vote       ``
+}
+
+type ValidVoter struct {
+	Email string `gorm:"primaryKey"`
+}
+
+type Candidate struct {
+	ID           uuid.UUID `gorm:"primaryKey"`
+	Name         string    `gorm:"not null"`
+	Presentation string    `gorm:"not null"`
+}
+
+type CastedVote struct {
+	VoterID    uuid.UUID  `gorm:"primaryKey"`
+	ElectionID uuid.UUID  `gorm:"primaryKey"`
+	Voter      ValidVoter `gorm:"foreignKey:VoterID"`
+	Election   Election   `gorm:"foreignKey:ElectionID"`
+}
+
+type Vote struct {
```

Comment:
Hur ser vi i denna vad rösten är lagd på?

Reply:
Right, väldigt sant, missade ett fält när jag gjorde om modellerna, ska finnas en candidate här också

Reply:
nvm that, saknar en hel table, ska vara ranking, så varje röst har flera kandidater de ska vara länkade till